### ISSUE TITLE
CharFilter with action lambda or named function is applied even without specifying it in url (django-filters + DRF)

### ISSUE THREAD

**Issue by elnygren** [alex/django-filter#279]
I stumbled on to a somewhat strange issue with the following code:

```
tag_regex = r' some regex here '
class MyFilter(django_filters.FilterSet):
  tags__have = django_filters.CharFilter(action = lambda q,v: q.filter(tags__regex = tag_regex.format(tag=v)))

  class Meta:
    model = ...
    nothing special here
```

a normal `GET` request without any filter parameters (I'm using django-filters with DRF ) would still apply the tags__have regex filter. Obviously I want the filter only to be applied when I do `GET /endpoint/?tags__have=something`.

If I do the same filter a bit differently with MethodFilter+lambda or MethodFilter+named function, this problem does not occur. I believe it is an issue of `CharFilter` and the `action` param. 

So in short, this problem can be avoided by using MethodFilter but it still seems like strange behaviour by CharFilter.

**Comment by carltongibson** [comment]
@elnygren Thanks for the report. I'm going to close this as I can't see an actionable issue per se.
